app/services/custom_fields/hierarchy/hierarchical_item_service.rb

```diff
@@ -83,7 +83,15 @@ def update_item(contract_class:, item:, label: nil, short: nil, score: nil)
       def delete_branch(item:)
         return Failure(:item_is_root) if item.root?
 
-        item.destroy ? Success() : Failure(item.errors)
+        # Remember information before destroying the item, self and descendants are needed as all of them are destroyed
+        item_ids = item.self_and_descendant_ids
+        custom_field = item.root&.custom_field
+        if item.destroy
+          update_calculated_values_for_hierarchy(item_ids:, custom_field:)
```

Comment:
Just a preference, but I think the `update_calculated_values_for_hierarchy` method arguments could be reduced to the item only.
```suggestion
        if item.destroy
          update_calculated_values_for_hierarchy(item)
```

Reply:
The comment then seems to be not clear - the problem with sending item itself, is that after destroying it `item.self_and_descendant_ids` will be empty and `item.root` will be nil, so they should be saved before and passed separately, I'll try to expand the comment